src/Compilers/CSharp/Portable/Symbols/PublicModel/AssemblySymbol.cs

```diff
@@ -81,9 +81,15 @@ bool IAssemblySymbol.GivesAccessTo(IAssemblySymbol assemblyWantingAccess)
 
                 AssemblyIdentity identity = UnderlyingAssemblySymbol.Identity;
 
+                // Avoid using the identity to obtain the public key if possible to avoid the allocations associated
+                // with identity creation
+                ImmutableArray<byte> publicKey = (assemblyWantingAccess is AssemblySymbol assemblyWantingAccessAssemblySymbol)
```

Comment:
>publicKey [](http://example.com/codeflow?start=37&length=9)

I think we should make a similar change for VB implementation as well

Reply:
Yep, good catch. Done in commit 2.